src/main/java/jenkins/plugins/git/GitSCMFileSystem.java

```diff
@@ -282,11 +282,16 @@ public SCMFileSystem build(@NonNull Item owner, @NonNull SCM scm, @CheckForNull
             if (rev != null && !(rev instanceof AbstractGitSCMSource.SCMRevisionImpl)) {
                 return null;
             }
-            TaskListener listener = new LogTaskListener(LOGGER, Level.FINE);
             GitSCM gitSCM = (GitSCM) scm;
             UserRemoteConfig config = gitSCM.getUserRemoteConfigs().get(0);
             BranchSpec branchSpec = gitSCM.getBranches().get(0);
             String remote = config.getUrl();
+            LogTaskListener listener = new LogTaskListener(LOGGER, Level.FINE);
+            if (remote == null) {
+                listener.getLogger().println("Git remote url is null");
+                listener.close();
```

Comment:
If close is needed here before the return from the method, shouldn't it also be done at every other return from the method?

Reply:
I think i did it because my IDE was warning on that. I'll remove that